### PR TITLE
new: syslog config

### PR DESCRIPTION
**Pull request by sibicramesh**
https://redlock.atlassian.net/browse/CNS-3670

### CODE REVIEW

File: specs/enforcerprofile.spec [hunk 1]
```diff
@@ -107,6 +107,58 @@ attributes:
     default_value: Docker
     deprecated: true
 
+  - name: syslogEnable
```
Comment (t00f) [inline specs/enforcerprofile.spec:110]: should be `syslogEnabled`

File: specs/enforcerprofile.spec [hunk 2]
```diff
@@ -107,6 +107,58 @@ attributes:
...
+    stored: true
+
+  - name: syslogEndpoint
+    description: Contains the remote endpoint to dispatch the syslog messages.
+    type: string
+    exposed: true
+    stored: true
+
+  - name: syslogEndpointTLSClientCertificate
+    description: PEM-encoded certificate to expose to the clients for TLS.
```
Comment (t00f) [inline specs/enforcerprofile.spec:123]: Not sure I understand the description - is it the certificate that will allow the enforcer to connect to the syslog server? What do we expose to clients?

File: specs/enforcerprofile.spec [hunk 3]
```diff
@@ -107,6 +107,58 @@ attributes:
...
+    exposed: true
+    stored: true
+
+  - name: syslogFacility
+    description: Contains the list of supported syslog facilities.
+    type: integer
+    exposed: true
+    stored: true
+    default_value: 1
+    max_value: 23
```
Comment (t00f) [inline specs/enforcerprofile.spec:149]: How is the user giving this value? Is it ok to ask for an integer? If we are limited, maybe we can have more human readable options?

File: specs/enforcerprofile.spec [hunk 4]
```diff
@@ -107,6 +107,58 @@ attributes:
...
+  - name: syslogEndpointTLSClientCertificateKey
+    description: |-
+      PEM-encoded certificate key associated with
+      `syslogEndpointTLSClientCertificate`.
+    type: string
+    exposed: true
+    stored: true
+    encrypted: true
+
+  - name: syslogEndpointTLSServerCA
```
Comment (t00f) [inline specs/enforcerprofile.spec:137]: Ok I feel like we want to do MTLS between the syslog server and the enforcer, right?

Comment (sibicramesh) [reply specs/enforcerprofile.spec:137]: Correct.

File: specs/enforcerprofile.spec (hunk 2 above)
Comment (sibicramesh) [reply specs/enforcerprofile.spec:123]: We will be doing MTLS.

File: specs/enforcerprofile.spec (hunk 3 above)
Comment (sibicramesh) [reply specs/enforcerprofile.spec:149]: @mheese wanted to have all the facilities and give the users the ability to choose from any of them.

File: specs/enforcerprofile.spec [hunk 5]
```diff
@@ -107,6 +107,58 @@ attributes:
...
+    stored: true
+
+  - name: syslogEndpointTLSClientCertificateKey
+    description: |-
+      PEM-encoded certificate key associated with
+      `syslogEndpointTLSClientCertificate`.
+    type: string
+    exposed: true
+    stored: true
+    encrypted: true
```
Comment (primalmotion) [inline specs/enforcerprofile.spec:135]: I believe this key needs to be sent to enforcer. We usually don't store private keys in a retrievable manner. This goes against the security posture of the backend.

**Review by primalmotion — CHANGES_REQUESTED** [review]

**Review by primalmotion — APPROVED** [review]

**Review by mheese — APPROVED** [review]
has everything we need from our end 👍

**Pull request merged**